plugins/aws-secrets-manager/src/static/startup/providers/secrets/secretsManager.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { SecretsManagerServiceBase } from "./base/secretsManager.service.base";
+
+@Injectable()
+export class SecretsManagerService extends SecretsManagerServiceBase {
+  constructor(
+    @Inject("AWS_SECRETS_MANAGER")
+    protected readonly secrets: Record<string, any>,
+    protected readonly configService: ConfigService
+  ) {
+    super(configService);
+  }
+
+  async getSecret<T>(key: string): Promise<T | null> {
```

Comment:
As you create the enum with all the secrets, it would be amazing if, when in STARTUP mode, the key type were `Secrets` (enum `Secrets`)

```suggestion
  async getSecret<T>(key: Secrets): Promise<T | null> {
```